shap/explainers/_deep/deep_tf.py

```diff
@@ -276,37 +328,54 @@ def shap_values(self, X, ranked_outputs=None, output_rank_order="max", check_add
                 model_output_ranks = np.argsort(np.abs(model_output_values))
             else:
                 assert False, "output_rank_order must be max, min, or max_abs!"
-            model_output_ranks = model_output_ranks[:,:ranked_outputs]
+            model_output_ranks = model_output_ranks[:, :ranked_outputs]
         else:
-            model_output_ranks = np.tile(np.arange(len(self.phi_symbolics)), (X[0].shape[0], 1))
+            model_output_ranks = np.tile(
+                np.arange(len(self.phi_symbolics)), (X[0].shape[0], 1)
+            )
 
         # compute the attributions
         output_phis = []
-        for i in range(model_output_ranks.shape[1]):
+        for output_index in range(model_output_ranks.shape[1]):
             phis = []
-            for k in range(len(X)):
-                phis.append(np.zeros(X[k].shape))
-            for j in range(X[0].shape[0]):
-                if (hasattr(self.data, '__call__')):
-                    bg_data = self.data([X[l][j] for l in range(len(X))])
+            # iterate over input tensors and prefill with zeroes
+            for input_index in range(len(X)):
+                phis.append(np.zeros(X[input_index].shape))
+
+            for sample_index in range(X[0].shape[0]):
+                if hasattr(self.data, "__call__"):
+                    bg_data = self.data([X[inp_idx][sample_index] for inp_idx in range(len(X))])
                     if type(bg_data) != list:
                         bg_data = [bg_data]
                 else:
                     bg_data = self.data
 
                 # tile the inputs to line up with the background data samples
-                tiled_X = [np.tile(X[l][j:j+1], (bg_data[l].shape[0],) + tuple([1 for k in range(len(X[l].shape)-1)])) for l in range(len(X))]
+                tiled_x = [
+                    np.broadcast_to(
+                        X[input_idx][sample_index],
+                        shape=bg_data[input_idx].shape
+                    )
+                    for input_idx in range(len(X))
```

Comment:
I changed this for me to better understand what was going on and hope this code it semantically the same but easier to read.